src/qt/coincontroldialog.cpp

```diff
@@ -600,6 +599,13 @@ void CoinControlDialog::updateLabels(WalletModel *model, QDialog* dialog)
         label->setVisible(nChange < 0);
 }
 
+CCoinControl* CoinControlDialog::coinControl()
+{
+    static CCoinControl* coin_control = nullptr;
+    if (!coin_control) coin_control = new CCoinControl();
```

Comment:
I think we leak this pointer?

Reply:
It was leaking already.

Reply:
But wouldn't that be a one-line change, to use `static unique_ptr<CCoinControl> coin_control;`? Or am I missing something?

Reply:
Do a Meyer's singleton

`
CCoinControl& CoinControlDialog::coinControl()
{
    static CCoinControl coin_control;
    return coin_control;
}
`

No pointers, no leaks.

Reply:
In that case when is coin_control instanced? I think the problem remains, but please correct me if I'm wrong.

Reply:
For now, pushed with `std::unique_ptr` as suggested by @laanwj.

Reply:
Thanks!

I think @kekimusmaximus's solution also works, static within a function is initialized on first use.

Reply:
That's new to me, I'll change to that instead.